From 06be0253e82f2b5057b3a4f2cb75644339606cca Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Mon, 29 Jan 2024 17:07:07 -0500
Subject: [PATCH 45/53] Add Samsung sysinput

---
 services/core/Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/Android.bp b/services/core/Android.bp
index 232fa6cb0831..3be9ab693335 100644
--- a/services/core/Android.bp
+++ b/services/core/Android.bp
@@ -214,6 +214,7 @@ java_library_static {
         "vendor.xiaomi.hardware.fingerprintextension-V1.0-java",
 
         //AIDL
+        "vendor.samsung.hardware.sysinput-V1-java",
         "vendor.samsung.hardware.biometrics.fingerprint-V1-java",
         "vendor.samsung.hardware.light-V1-java",
     ],
-- 
2.34.1

