From 244cd163345c486cabd8e15a358a1f64760413e6 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Mon, 29 Jan 2024 17:07:07 -0500
Subject: [PATCH 52/56] Add Samsung sysinput

---
 services/core/Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/Android.bp b/services/core/Android.bp
index 6cba27c50972..0e822cfd5abe 100644
--- a/services/core/Android.bp
+++ b/services/core/Android.bp
@@ -196,6 +196,7 @@ java_library_static {
         "vendor.xiaomi.hardware.fingerprintextension-V1.0-java",
 
         //AIDL
+        "vendor.samsung.hardware.sysinput-V1-java",
         "vendor.samsung.hardware.biometrics.fingerprint-V1-java",
         "vendor.samsung.hardware.light-V1-java",
     ],
-- 
2.34.1

