From cb79745f2052faa461db44b9189fafa7cca3f4fe Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Mon, 29 Nov 2021 17:49:13 -0500
Subject: [PATCH 7/7] Don't unmount rw-system.sh binds

Change-Id: If9132c21defa8b09879b79a70794c5275d6852d0
---
 VolumeManager.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/VolumeManager.cpp b/VolumeManager.cpp
index 2697f67..d592c7f 100644
--- a/VolumeManager.cpp
+++ b/VolumeManager.cpp
@@ -957,6 +957,7 @@ int VolumeManager::unmountAll() {
     while ((mentry = getmntent(fp)) != NULL) {
         auto test = std::string(mentry->mnt_dir);
         if ((StartsWith(test, "/mnt/") &&
+             !StartsWith(test, "/mnt/phh") &&
 #ifdef __ANDROID_DEBUGGABLE__
              !StartsWith(test, "/mnt/scratch") &&
 #endif
-- 
2.25.1

