From 2a51f1ffb1a0d9769d846669029435ea5350dca9 Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Fri, 10 Nov 2023 00:58:32 +0000
Subject: [PATCH 1/2] Make ThemePicker override Live Wallpapers picker too

---
 Android.bp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Android.bp b/Android.bp
index 2aa5eb74..1fb20b8d 100644
--- a/Android.bp
+++ b/Android.bp
@@ -128,5 +128,5 @@ android_app {
     platform_apis: true,
     manifest: "AndroidManifest.xml",
     additional_manifests: [":WallpaperPicker2_Manifest"],
-    overrides: ["WallpaperCropper", "WallpaperPicker", "WallpaperPicker2"],
+    overrides: ["LiveWallpapersPicker", "WallpaperCropper", "WallpaperPicker", "WallpaperPicker2"],
 }
-- 
2.34.1

