From 231447df3b3c36e45410bbcf7f79aca05c69cfbe Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Tue, 7 May 2024 22:44:25 +0000
Subject: [PATCH 1/2] Dont abort if the DnsHelper failed to init on BPF-less
 kernel

---
 DnsProxyListener.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/DnsProxyListener.cpp b/DnsProxyListener.cpp
index 04c59212..3d260966 100644
--- a/DnsProxyListener.cpp
+++ b/DnsProxyListener.cpp
@@ -691,7 +691,6 @@ IsUidBlockedFn resolveIsUidNetworkingBlockedFn() {
     const int ret = (*ADnsHelper_init)();
     if (ret) {
         LOG(ERROR) << __func__ << ": ADnsHelper_init failed " << strerror(-ret);
-        abort();
     }
 
     IsUidBlockedFn f =
-- 
2.34.1

