From fa7acbe8418c6f2c91af61c6b8f715fc0f8b5ab5 Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Mon, 11 Oct 2021 22:34:14 +0100
Subject: [PATCH 1/2] Disable library check on TrebleApp build

---
 TrebleApp/Android.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/TrebleApp/Android.mk b/TrebleApp/Android.mk
index 5acae71..daba54d 100644
--- a/TrebleApp/Android.mk
+++ b/TrebleApp/Android.mk
@@ -9,5 +9,6 @@ LOCAL_MODULE_SUFFIX := .apk
 LOCAL_BUILT_MODULE_STEM := package.apk
 LOCAL_CERTIFICATE := platform
 LOCAL_PRIVILEGED_MODULE := true
+LOCAL_ENFORCE_USES_LIBRARIES := false
 
 include $(BUILD_PREBUILT)
-- 
2.25.1

