From 70073cb180051a0bcedb5fac624e9b9927faefc3 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 14 Dec 2019 23:46:34 +0200
Subject: [PATCH 1/2] Remove su from vanilla builds

---
 target/product/base_system.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/target/product/base_system.mk b/target/product/base_system.mk
index c919ac462b..6d149b89e6 100644
--- a/target/product/base_system.mk
+++ b/target/product/base_system.mk
@@ -388,7 +388,6 @@ PRODUCT_PACKAGES_DEBUG := \
     ss \
     start_with_lockagent \
     strace \
-    su \
     sanitizer-status \
     tracepath \
     tracepath6 \
-- 
2.25.1

