From c046f10027f788fc6645a7d377f0c9e90ce399e6 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 14 Dec 2019 23:46:34 +0200
Subject: [PATCH 1/2] Remove su from vanilla builds

---
 target/product/base_system.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/target/product/base_system.mk b/target/product/base_system.mk
index c6ca3dda2c..58d9629a92 100644
--- a/target/product/base_system.mk
+++ b/target/product/base_system.mk
@@ -436,7 +436,6 @@ PRODUCT_PACKAGES_DEBUG := \
     ss \
     start_with_lockagent \
     strace \
-    su \
     sanitizer-status \
     tracepath \
     tracepath6 \
-- 
2.34.1

