From 16d99864d71b606d183b118d28bdeb09304181e9 Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Wed, 13 Oct 2021 00:01:50 +0100
Subject: [PATCH] gapps.mk: update PE commands

---
 gapps.mk | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/gapps.mk b/gapps.mk
index 15e12a9..0f161da 100644
--- a/gapps.mk
+++ b/gapps.mk
@@ -43,10 +43,12 @@ endif
 
 #Pixel Experience gapps
 ifneq ($(wildcard vendor/gapps),)
-IS_PHONE := true
-$(call inherit-product,vendor/gapps/config.mk)
+TARGET_SUPPORTS_GOOGLE_RECORDER ?= true
+TARGET_INCLUDE_STOCK_ARCORE ?= true
+TARGET_INCLUDE_LIVE_WALLPAPERS ?= true
+ifeq ($(TARGET_INCLUDE_LIVE_WALLPAPERS),true)
 PRODUCT_PACKAGES += \
-       phh-overrides \
-	   com.simplemobiletools.gallery.pro \
-
+    PixelLiveWallpapersOverlay
+endif
+$(call inherit-product,vendor/gapps/config.mk)
 endif
-- 
2.25.1

