From 3f43ab251a7d2e04f6e4df755e0f85c16410b617 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Fri, 6 Oct 2023 19:11:22 -0400
Subject: [PATCH 5/6] Include vndk v28 sepolicy support

---
 core/config.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/config.mk b/core/config.mk
index 0c086ee0907..6006a5e8f5f 100644
--- a/core/config.mk
+++ b/core/config.mk
@@ -911,6 +911,7 @@ endif
 
 # A list of SEPolicy versions, besides PLATFORM_SEPOLICY_VERSION, that the framework supports.
 PLATFORM_SEPOLICY_COMPAT_VERSIONS := \
+    28.0 \
     29.0 \
     30.0 \
     31.0 \
-- 
2.34.1

