From b9c1802b65bea0c63dd2063b229945015dfecc4c Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Sat, 16 Oct 2021 14:24:09 +0100
Subject: [PATCH] gapps: update gapps commands

---
 gapps.mk | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/gapps.mk b/gapps.mk
index 15e12a9..d771dcb 100644
--- a/gapps.mk
+++ b/gapps.mk
@@ -43,10 +43,12 @@ endif
 
 #Pixel Experience gapps
 ifneq ($(wildcard vendor/gapps),)
-IS_PHONE := true
+TARGET_SUPPORTS_GOOGLE_RECORDER ?= true
+TARGET_INCLUDE_STOCK_ARCORE ?= true
+TARGET_INCLUDE_LIVE_WALLPAPERS ?= true
+ifeq ($(TARGET_INCLUDE_LIVE_WALLPAPERS),true)
+ PRODUCT_PACKAGES += \
+    PixelLiveWallpapersOverlay
+endif
 $(call inherit-product,vendor/gapps/config.mk)
-PRODUCT_PACKAGES += \
-       phh-overrides \
-	   com.simplemobiletools.gallery.pro \
-
 endif
-- 
2.25.1

