From ab391255baeac3660c9eb6f0556bf897669c7246 Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Wed, 25 Dec 2019 12:34:05 +0200
Subject: [PATCH 2/3] base: securize system

	modified:   base.mk
---
 base.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base.mk b/base.mk
index 3697c49..c1659ab 100644
--- a/base.mk
+++ b/base.mk
@@ -161,7 +161,7 @@ PRODUCT_PACKAGES += \
 	resetprop_phh
 
 PRODUCT_COPY_FILES += \
-	device/phh/treble/phh-securize.sh:system/bin/phh-securize.sh \
+	device/phh/treble/empty:system/phh/secure \
 	device/phh/treble/files/ota.sh:system/bin/ota.sh \
 
 PRODUCT_COPY_FILES += \
-- 
2.25.1

