From f122a0b9671501c0b45c4251d7217f607e8f8509 Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Fri, 12 Apr 2024 23:59:25 +0000
Subject: [PATCH 3/3] a64: Set BUILD_BROKEN_MISSING_REQUIRED_MODULES to true

---
 tdgsi_a64_ab/BoardConfig.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tdgsi_a64_ab/BoardConfig.mk b/tdgsi_a64_ab/BoardConfig.mk
index f131e88..72c4c73 100644
--- a/tdgsi_a64_ab/BoardConfig.mk
+++ b/tdgsi_a64_ab/BoardConfig.mk
@@ -8,3 +8,4 @@ ifeq ($(BOARD_SYSTEMIMAGE_PARTITION_RESERVED_SIZE),)
 BOARD_SYSTEMIMAGE_PARTITION_SIZE := 1572864000
 endif
 TARGET_USES_64_BIT_BINDER := true
+BUILD_BROKEN_MISSING_REQUIRED_MODULES := true
-- 
2.34.1

