From 19311ffc518f687688dd8e6a72decddba6e6d4cf Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Tue, 3 Dec 2019 14:04:17 +0100
Subject: [PATCH 15/40] Fix for some Huawei camera

---
 services/camera/libcameraservice/CameraService.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index e56309b466..e02626fcda 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -16,7 +16,7 @@
 
 #define LOG_TAG "CameraService"
 #define ATRACE_TAG ATRACE_TAG_CAMERA
-//#define LOG_NDEBUG 0
+#define LOG_NDEBUG 0
 
 #include <algorithm>
 #include <climits>
@@ -235,10 +235,11 @@ status_t CameraService::enumerateProviders() {
             if (!cameraFound) {
                 addStates(id8);
             }
-        }
-
-        if (getCameraState(id8) == nullptr) {
             onDeviceStatusChanged(id8, CameraDeviceStatus::PRESENT);
+        } else {
+            if (getCameraState(id8) == nullptr) {
+                onDeviceStatusChanged(id8, CameraDeviceStatus::PRESENT);
+            }
         }
     }
 
-- 
2.25.1

