From 8ad70804f1f389335f14d520a3d597e5e25e112e Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Mon, 11 Oct 2021 17:32:42 +0100
Subject: [PATCH] A12 uncommitted fixes

---
 media/libaudiohal/impl/Android.bp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/media/libaudiohal/impl/Android.bp b/media/libaudiohal/impl/Android.bp
index f8f144c14c..4e360be7e8 100644
--- a/frameworks/av/media/libaudiohal/impl/Android.bp
+++ b/frameworks/av/media/libaudiohal/impl/Android.bp
@@ -69,7 +69,9 @@ cc_library_shared {
         "android.hardware.audio.common@2.0",
         "android.hardware.audio.common@2.0-util",
         "android.hardware.audio.effect@2.0",
+        "android.hardware.audio.effect@2.0-util",
         "android.hardware.audio@2.0",
+        "android.hardware.audio@2.0-util",
     ],
     cflags: [
         "-DMAJOR_VERSION=2",
-- 
2.25.1

